### PR TITLE
Add suspend/resume relation support

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

Cross model relations to application offers can be suspended/resumed.
$ juju suspend-relation 123
$ juju resume-relation 123

A suspended relation will have the departed/broken hooks run but the relation remains in the model with the status "suspended". Similarly, a resumed relation will run the joined/changed hooks and the relation status becomes "joined".

## QA steps

Deploy a CMR scenario, mediawiki, mysql
On mysql model:
$ juju suspend-relation 1
verify that relation breaks, hooks are run using show-status-log, check list-offers shows suspended
$ juju resume-relation 1
verify that relation is repaired, hooks are run, list-offers shows joined

### CODE REVIEW

File: cmd/juju/application/resumerelation.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/api/application"
+	"github.com/juju/juju/cmd/juju/block"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/core/relation"
+)
+
+var resumeHelpSummary = `
+Resumes a relation to an application offer.`[1:]
```
Comment (babbageclunk) [inline cmd/juju/application/resumerelation.go]: Maybe `Resumes a suspended relation...`?

File: worker/uniter/relation/relations.go [hunk 2]
```diff
@@ -202,14 +202,15 @@ func (r *relations) NextHook(
...
 		var remoteBroken bool
-		if remoteState.Life == params.Dying || relationSnapshot.Life == params.Dying {
+		if remoteState.Life == params.Dying ||
+			relationSnapshot.Life == params.Dying || relationSnapshot.Status == params.Suspended {
 			relationSnapshot = remotestate.RelationSnapshot{}
 			remoteBroken = true
 			// TODO(axw) if relation is implicit, leave scope & remove.
 		}
-		// If either the unit or the relation are Dying,
+		// If either the unit or the relation are Dying, or the relation becomes revoking,
```
Comment (babbageclunk) [inline worker/uniter/relation/relations.go]: suspended

File: cmd/juju/application/resumerelation.go (hunk 1 above)
Comment (wallyworld) [reply cmd/juju/application/resumerelation.go]: fixed

File: worker/uniter/relation/relations.go (hunk 2 above)
Comment (wallyworld) [reply worker/uniter/relation/relations.go]: fixed

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**